2018 Robot/src/org/usfirst/frc/team4999/robot/sensors/VMXPi.java

```diff
@@ -0,0 +1,121 @@
+package org.usfirst.frc.team4999.robot.sensors;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.nio.ByteBuffer;
+
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+/**
+ *
+ */
+public class VMXPi implements PIDSource {
+
+    private static int PORT = 5800;
+	private DatagramSocket server;
+	DatagramPacket packet = new DatagramPacket(new byte[17], 17);
```

Comment:
I count at least 3 instances of the number 17 hardcoded.  Must be a shared constant.